tycho-its/src/test/java/org/eclipse/tycho/test/product/ProductBuildTest.java

```diff
@@ -9,17 +9,48 @@
  *******************************************************************************/
 package org.eclipse.tycho.test.product;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
 import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 import org.apache.maven.it.Verifier;
+import org.eclipse.tycho.TychoConstants;
 import org.eclipse.tycho.test.AbstractTychoIntegrationTest;
 import org.junit.Test;
 
+import de.pdark.decentxml.Document;
+import de.pdark.decentxml.Element;
+import de.pdark.decentxml.XMLParser;
+
 public class ProductBuildTest extends AbstractTychoIntegrationTest {
 	@Test
 	public void testMavenDepedencyInTarget() throws Exception {
 		Verifier verifier = getVerifier("product.mavenLocation", false);
 		verifier.executeGoals(Arrays.asList("clean", "verify"));
 		verifier.verifyErrorFreeLog();
 	}
+
+	@Test
+	public void testPGPSignedInProduct() throws Exception {
+		Verifier verifier = getVerifier("product.pgp", false);
+		verifier.executeGoals(Arrays.asList("clean", "install"));
+		verifier.verifyErrorFreeLog();
+		File artifactXml = new File(verifier.getBasedir(), "target/repository/artifacts.xml");
```

Comment:
This should test the artifacts.xml of the product not of the repository.  I.e., in
```
target/products/org.eclipse.oomph.setup.installer.product/win32/win32/x86_64/artifacts.xml
```
And it should check that `pgp.publicKeys` is also present because that's specifically what the about dialog looks at and that property is specifically populated when the signature is verified during installation.

Reply:
The `artifacts.xml` is generated, and the build now fails with NPE on 2.7.0, but the artifacts is mostly empty, is this intended?

```
<artifact classifier='osgi.bundle' id='org.mockito.mockito-core' version='4.1.0'>
  <properties size='1'>
    <property name='download.size' value='670719'/>
  </properties>
</artifact>
```

Reply:
sorry it is not empty, but the properties are empty (see the snippet above), but maybe this is normal behavior?

Reply:
It's not normal, it's bad.  It is the topic of this issue when it was the case for the platform's SDK product:

https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/132

Download any 4.24 product such as this one:

https://download.eclipse.org/eclipse/downloads/drops4/I20220331-1800/download.php?dropFile=eclipse-SDK-I20220331-1800-win32-x86_64.zip

And look in the artifacts.xml for any of the PGP-signed artifacts and they look like this:

```
    <artifact classifier='osgi.bundle' id='org.eclipse.jetty.servlet' version='10.0.8'>
      <properties size='3'>
        <property name='pgp.signatures' value='-----BEGIN PGP SIGNATURE-----&#xA;Version: GnuPG v2.0.22 (GNU/Linux)&#xA;&#xA;iQIcBAABAgAGBQJiRjOoAAoJEHAOTzm8BTZLyNEQAIWpRD/c6o2M+t2SJddMzXgf&#xA;XXwhr39tZuuqOyz+tN6Fvmd8nuHz1N9OjRrnv3zgYNzUzb8a+sIkpr8ndiRJQauJ&#xA;iO8jjfKbRtk0BsAtZJRDTB+2M/Y1KQAq+3vlxD5WigAMLOfhP5/yjO5xXG8WAj2v&#xA;Z8HxG7covlfLg4YfBK8/MHkHUK73qHmxUjmhYAcj3mEwLFx73tg342sV04uA3w2f&#xA;auQN3iPH6PzjrseRsgSmQUviVr2pwzTvy/t//wahM9/NQKUjhQ/LxmCsb0Lc5V3v&#xA;561tLAuUtUZEkI4E5kRi62oha9OTTJUvAt9d2yfYsAJm7aZ/JybAYsTViH7gLG9e&#xA;Q74V9f8mW8ttfxiHJKhg86T14FF+yBncu0p1EbAqXOyOuD3+XbveB7roP1Ur1nZR&#xA;huUctkAcaGODOY8gJ1RS/nOERKPfHudpvRthKoiWUgZvGtdUvu4CHVk1OV7oz2sa&#xA;mvniKZqhwJLjm49KQIP9o6Ij/Mdzat1vBF/aMnMZq6rMwt/wTtv7cWusX8X1yenw&#xA;WS7sgsyoZYFp4cwTB39duuYBwMhiHRULNavWWkVhM4qujjBmp8II2/iHF7V8s6Ek&#xA;CGgHxUIwDY3Y/KdOsZV+oEw0kLk5+doyi5KCIIpeKlDQ6H87vdCUrzMmMJ5DvYCZ&#xA;TYaDZOfk3tB+2DEpJgL2&#xA;=Cb6t&#xA;-----END PGP SIGNATURE-----'/>
        <property name='pgp.publicKeys' value='-----BEGIN PGP MESSAGE-----&#xA;Version: BCPG v1.70&#xA;&#xA;uQINBFhaXPsBEAC3bR7f5euHbpIDDTuFYHPI0+S5X0DhuqcGBUL2HSFhWMwIlfsA&#xA;aO+pt7GyfXLUkTmzugwmwO+sOW2QmwEZQcK2z3BrcjytZophZ9AUajbAjnadSH6U&#xA;XCMmfExVVnaYSfl/+Uub42szQE/r3gCRIz6M6clVVAjpFv4G/mumfQUV/XzLoUEY&#xA;XTgwTokFJ97R+hDbHvBEBrUT8M6zHP5DhN3EBug3qb6wZVOa/+HEX3M+7k4jVT/p&#xA;pNumw0acg0DDoSNQ13VsRV6sV0XE4zr3Zfs84f8xCgXpEMs4U6DZGqs3iJVVtbRf&#xA;0oL0fgcxNgRrmbCrBfbXYfrS4u+fJ0vB+Wrflv9eNA3i6TtVL6uYpZy9uO2B1olK&#xA;VzfEhsgB3QrULB4jVHZjIXGe4ILn45ndMtAeY4M91wyobgG99Xl+1vPHrxV0+2zR&#xA;P66J3puyxiKE2B7gd7hib54CB3lYyrG1S+K1kZGCI1IFKCnqmTJXY0tKoLAASS3v&#xA;tDcknXenzR5RVSpWTDuxtusekfL0Bw8pCBoz9L4Hex8Q1j//D5CZlqcg1NKFfmBZ&#xA;7ta9PTuJcpOsz/LaPG/0VHYt/QAv5o4eeZESl7iZyM4/0NFh2s/rq0R8Z9yVSSkI&#xA;vvO8d8XGZ65NTm3T4NFuEihn+AEm+zg4KiGdYBEZvs8QQoW9e1+MMN8xnwARAQAB&#xA;iQREBBgBCAAPAhsCBQJhuzR9BQkSxtkCAinBXSAEGQEIAAYFAlhaXPsACgkQcA5P&#xA;ObwFNkunSw//SRR1tGS1pDj2jonLpR0wPilCphS6ANv895yvlg6rHG4nKi4hQ0Jz&#xA;ZxhGCwkgxEkRaKiyLfEiTihETkF161AqLPhyvE8LuQ1AG+A+tUnR8/T3gKE8t/m2&#xA;/UtScZwN1QEQVc/uG7MTrbZ2ngXfH65k3fzhjy95AnJHAswu2vic1hzDi77HlQpN&#xA;0O3adJuU/jfdu1RxNE0MRt8MFEjsTFwSBVm6lDxgcZV+qjRLGQznTyLF5/AyCI7Z&#xA;4z9xHZPKFq1eHzqevifNiqfb8KX22sHKOSdnVBzBq/UxbT5jIbNSRhD91FjtZD7Z&#xA;6wi3POsB/9RWZBldCov4ZEajmxFzxpx4RAqYOSIkEor9ZtRGbZuWvTie4vFIur7T&#xA;f543mE6nxKcggExNp4MTyOd1scMc9oyczH561OTdHOCYEyoCwpG9N2Hb1/MDnWSi&#xA;HKG451CvdrE5FHcPZKjp/nHUcRw/WQC3bgj6ScAay64EKC5S9tW+Wp85Oyyvj+M7&#xA;lBzOxp19nESpfC++fzBAQPMxtD8EvrZTxqFSJxMOH9bhzB8+MFt08tmYb5SwoYi4&#xA;C8JJ+wZgNetJKK+j07fvyMUChH/SbkCVszMiiSEjHA2Kk0LMVYKS/OLJU7i7tZXV&#xA;aJ078QEeTDy5hSzsutd+orlFkR9+mgr1HUh0UgYlofTfEi7bLDeSr0cJELbTq5vM&#xA;ZBKCicIP/irazYBVKw0SluhHtjzRcs5WIdH5bVPsEE87+iUc4daONWdVIhLdokxt&#xA;OWlrEmZFLKqq9Z8fzvlf5LAQMOBkMAkl0z2ej4KG7zrjWyqDgysEI2WBlqTAFSeL&#xA;+89Kc9BzJE9heYW8EfpXbNfOnKnAYWsbhcomSxVQ/jBIuyLBg/0gYKpBNx8HC6v9&#xA;xNH0Ja+wM/7w3JC1aIwMYJn1yF2ykUYS+BoTCU7TA8r43pHg4I4Fz+Y2P5RLk+RJ&#xA;I4kJezDNiJOpIcr/nKTPxMGUzMtWlGyAJ7LkyOZCtQXhtXwaT8grjtHzlwlGrpgD&#xA;Rtf7wWjzEWeaQSegTFM9Mid+09kCp0PkJvveg8wJCuoVboNOto0O5rQsUczjXxiW&#xA;kXYlHGeQL4rWc1zP7F1n4DEwDbVZC7jOn/80l3x4LcKuhc86gP4L5HKbdjn5GcQ0&#xA;3RVLl1WVTQCdpr0+am28hl9XpyHdlWwSEmqqoUnjGv5B8RClocBRS4ECPPZCVSBl&#xA;yK8eDgRww9Fu1EFq4xkq5fGj4YUOAIm756iW41NQ3VnPYbom/J27iFFN8+h92CSb&#xA;KAqhmRwQh+GGo0eGCXmPHyQ/KCHTvnTZCFBUvabm3rVNFaDO+RvmwPwNCRz0DYzG&#xA;paeMOGo4nMMGbzdhgfJ/X5Ed1/Mqz8egHhGIO94ebKEN5ZtJjAOK&#xA;=xtSl&#xA;-----END PGP MESSAGE-----&#xA;'/>
        <property name='download.size' value='163246'/>
      </properties>
    </artifact>
```
This is the result that's needed and is what Tycho 3.0.0 produces...